<!doctype html><html lang=en-us dir=ltr><head><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="
  NTU Machine Learning Final Project Proposal Notes
  #


  tags: NTU_ML Machine Learning
  #


  Deep6mAPred: A CNN and Bi-LSTM-based deep learning method for predicting DNA N6-methyladenosine sites across plant species
  #


  Introduction & Motivation
  #


DNA methylation alters activities of DNA segments without changing the sequence, which thus yields a wide variety of roles in the cellular processes across organisms or tissues
DNA methylation is widely distributed both in prokaryote and in eukaryote, but the proportion of methylated residues differs greatly with species
DNA methylation is essential for normal development
DNA methylation is increasingly attracting attentions from biologists


  Related works
  #



Even Luo et al.[41] - proposed the DNA 6mA as a new epigenetic mark in eukaryotes"><meta name=theme-color media="(prefers-color-scheme: light)" content="#ffffff"><meta name=theme-color media="(prefers-color-scheme: dark)" content="#343a40"><meta name=color-scheme content="light dark"><meta property="og:url" content="https://bernie6401.github.io/security/course/ntu-ml/ntu-machine-learning-final-project-proposal-notes/"><meta property="og:site_name" content="SBK Hugo Site"><meta property="og:title" content="NTU Machine Learning Final Project Proposal Notes"><meta property="og:description" content="NTU Machine Learning Final Project Proposal Notes # tags: NTU_ML Machine Learning # Deep6mAPred: A CNN and Bi-LSTM-based deep learning method for predicting DNA N6-methyladenosine sites across plant species # Introduction & Motivation # DNA methylation alters activities of DNA segments without changing the sequence, which thus yields a wide variety of roles in the cellular processes across organisms or tissues DNA methylation is widely distributed both in prokaryote and in eukaryote, but the proportion of methylated residues differs greatly with species DNA methylation is essential for normal development DNA methylation is increasingly attracting attentions from biologists Related works # Even Luo et al.[41] - proposed the DNA 6mA as a new epigenetic mark in eukaryotes"><meta property="og:locale" content="en_us"><meta property="og:type" content="article"><meta property="article:section" content="security"><meta property="article:tag" content="NTU_ML"><meta property="article:tag" content="Machine Learning"><meta property="article:tag" content="NTU"><title>NTU Machine Learning Final Project Proposal Notes | SBK Hugo Site</title>
<link rel=icon href=/favicon.png><link rel=manifest href=/manifest.json><link rel=canonical href=https://bernie6401.github.io/security/course/ntu-ml/ntu-machine-learning-final-project-proposal-notes/><link rel=stylesheet href=/book.min.6c8b9d2a1fc95075ed7da46ca81060b39add8fff6741ac51259f768929281e2c.css integrity="sha256-bIudKh/JUHXtfaRsqBBgs5rdj/9nQaxRJZ92iSkoHiw=" crossorigin=anonymous><script defer src=/fuse.min.js></script><script defer src=/en.search.min.acdc41c8d39e6c69d70d8a23779875e0a3733fefead3e428d5344966bb12f562.js integrity="sha256-rNxByNOebGnXDYojd5h14KNzP+/q0+Qo1TRJZrsS9WI=" crossorigin=anonymous></script></head><body dir=ltr><input type=checkbox class="hidden toggle" id=menu-control>
<input type=checkbox class="hidden toggle" id=toc-control><main class="container flex"><aside class=book-menu><div class=book-menu-content><nav><h2 class=book-brand><a class="flex align-center" href=/><span>SBK Hugo Site</span></a></h2><div class="book-search hidden"><input type=text id=book-search-input placeholder=Search aria-label=Search maxlength=64 data-hotkeys=s/><div class="book-search-spinner hidden"></div><ul id=book-search-results></ul></div><script>document.querySelector(".book-search").classList.remove("hidden")</script></nav><script>(function(){var e=document.querySelector("aside .book-menu-content");addEventListener("beforeunload",function(){localStorage.setItem("menu.scrollTop",e.scrollTop)}),e.scrollTop=localStorage.getItem("menu.scrollTop")})()</script></div></aside><div class=book-page><header class=book-header><div class="flex align-center justify-between"><label for=menu-control><img src=/svg/menu.svg class=book-icon alt=Menu></label><h3>NTU Machine Learning Final Project Proposal Notes</h3><label for=toc-control><img src=/svg/toc.svg class=book-icon alt="Table of Contents"></label></div><aside class="hidden clearfix"><nav id=TableOfContents><ul><li><ul><li></li></ul></li><li><a href=#deep6mapred-a-cnn-and-bi-lstm-based-deep-learning-method-for-predicting-dna-n6-methyladenosine-sites-across-plant-species>Deep6mAPred: A CNN and Bi-LSTM-based deep learning method for predicting DNA N6-methyladenosine sites across plant species</a><ul><li><a href=#introduction--motivation>Introduction & Motivation</a></li><li><a href=#related-works>Related works</a></li><li><a href=#background>Background</a></li><li><a href=#my-opinions>My Opinions</a></li><li><a href=#comparison>Comparison</a></li><li><a href=#other-issue>Other Issue</a></li><li><a href=#conclusion>Conclusion</a></li><li><a href=#references>References</a></li></ul></li><li><a href=#ensemble-learning-of-convolutional-neural-network-support-vector-machine-and-best-linear-unbiased-predictor-for-brain-age-prediction-aramis-contribution-to-the-predictive-analytics-competition-2019-challenge>Ensemble Learning of Convolutional Neural Network, Support Vector Machine, and Best Linear Unbiased Predictor for Brain Age Prediction: ARAMIS Contribution to the Predictive Analytics Competition 2019 Challenge</a><ul><li><a href=#introduction--motivation-1>Introduction & Motivation</a></li><li><a href=#related-works-1>Related works</a></li><li><a href=#background-1>Background</a></li><li><a href=#my-opinions-1>My Opinions</a></li><li><a href=#comparison-1>Comparison</a></li><li><a href=#other-issue-1>Other Issue</a></li><li><a href=#conclusion-1>Conclusion</a></li><li><a href=#references-1>References</a></li></ul></li><li><a href=#machine-learning-workflows-to-estimate-class-probabilities-for-precision-cancer-diagnostics-on-dna-methylation-microarray-data>Machine learning workflows to estimate class probabilities for precision cancer diagnostics on DNA methylation microarray data</a><ul><li><a href=#introduction--motivation-2>Introduction & Motivation</a></li><li><a href=#related-works-2>Related works</a></li><li><a href=#background-2>Background</a></li><li><a href=#my-opinions-2>My Opinions</a></li><li><a href=#comparison-2>Comparison</a></li><li><a href=#other-issue-2>Other Issue</a></li><li><a href=#conclusion-2>Conclusion</a></li><li><a href=#references-2>References</a></li></ul></li></ul></nav></aside></header><article class="markdown book-article"><h1 id=ntu-machine-learning-final-project-proposal-notes>NTU Machine Learning Final Project Proposal Notes
<a class=anchor href=#ntu-machine-learning-final-project-proposal-notes>#</a></h1><h6 id=tags-ntu_ml-machine-learning>tags: <code>NTU_ML</code> <code>Machine Learning</code>
<a class=anchor href=#tags-ntu_ml-machine-learning>#</a></h6><h2 id=deep6mapred-a-cnn-and-bi-lstm-based-deep-learning-method-for-predicting-dna-n6-methyladenosine-sites-across-plant-species>Deep6mAPred: A CNN and Bi-LSTM-based deep learning method for predicting DNA N6-methyladenosine sites across plant species
<a class=anchor href=#deep6mapred-a-cnn-and-bi-lstm-based-deep-learning-method-for-predicting-dna-n6-methyladenosine-sites-across-plant-species>#</a></h2><h3 id=introduction--motivation>Introduction & Motivation
<a class=anchor href=#introduction--motivation>#</a></h3><ul><li>DNA methylation alters activities of DNA segments without changing the sequence, which thus yields a wide variety of roles in the cellular processes across organisms or tissues</li><li>DNA methylation is widely distributed both in prokaryote and in eukaryote, but the proportion of methylated residues differs greatly with species</li><li>DNA methylation is essential for normal development</li><li>DNA methylation is increasingly attracting attentions from biologists</li></ul><h3 id=related-works>Related works
<a class=anchor href=#related-works>#</a></h3><ul><li><p>Even Luo et al.[41] - proposed the DNA 6mA as a new epigenetic mark in eukaryotes</p></li><li><p>2 major ways to detect DNA 6mA sites:</p><ul><li>Wet experiments<ul><li>mass spectrometry / methylation-specific polymerase chain reaction / single-molecule real-time sequencing</li><li>Times consuming / expensive / labor-intensive</li></ul></li><li>Dry experiments - computational methods<ul><li>learning a classifier: feature-based and the deep learning-based methods</li></ul></li><li>Pian et al.[45] - employed transition probability between adjacent nucleotides to construct a Markov model to identify 6mA site</li><li>Chen et al.[47] – <code>SVM</code> based method for rice 6mA prediction</li><li>The i6mA-DNCP[51] - improve the representation with optimized dinucleotide-based features</li><li>iDNA6mA-Rice[1] – multiple encoding schemes as representations / random forest</li><li>6mA-RicePred[53] – fused multiple features including Markov feature for rice 6mA prediction</li><li>csDMA[54] – three representations and explore the best one</li><li>6mA-Finder[55] – seven sequence-based coding schemes</li><li>iIM-CNN - employed CNN</li><li>iDNA6mA - deep learning-based method / CNN</li><li>Huang et al. [72] - LSTM-based method called 6mA-Pred</li><li>Le et al. [73] - employed transformer-based BERT</li><li>DNA6mA-MINT [74] - stacked CNN and LSTM simply</li></ul></li><li><p>Appendix</p><ul><li>[41]G.Z. Luo, M.A. Blanco, E.L. Greer, C. He, Y. Shi, DNA N(6)-methyladenine: a new epigenetic mark in eukaryotes? Nat. Rev. Mol. Cell Biol. 16 (12) (2015) 705–710, <a href=https://doi.org/10.1038/nrm4076>https://doi.org/10.1038/nrm4076</a>.</li><li>[45] C. Pian, G. Zhang, F. Li, X. Fan, MM-6mAPred: identifying DNA N6- methyladenine sites based on Markov model, Bioinformatics 36 (2) (2020) 388–392, <a href=https://doi.org/10.1093/bioinformatics/btz556>https://doi.org/10.1093/bioinformatics/btz556</a>.</li><li>[47] W. Chen, H. Lv, F. Nie, H. Lin, i6mA-Pred: identifying DNA N6-methyladenine sites in the rice genome, Bioinformatics 35 (16) (2019) 2796–2800, <a href=https://doi.org/10.1093/bioinformatics/btz015>https://doi.org/10.1093/bioinformatics/btz015</a>.</li><li>[51] L. Kong, L. Zhang, i6mA-DNCP: computational identification of DNA N6- methyladenine sites in the rice genome using optimized dinucleotide-based features, Genes 10 (10) (2019) 828, <a href=https://doi.org/10.3390/genes10100828>https://doi.org/10.3390/genes10100828</a>.</li><li>[1] H. Lv, F.Y. Dao, Z.X. Guan, D. Zhang, J.X. Tan, Y. Zhang, W. Chen, H. Lin, iDNA6mA-Rice: a computational tool for detecting N6-methyladenine sites in rice, Front. Genet. 10 (2019) 793, <a href=https://doi.org/10.3389/fgene.2019.00793>https://doi.org/10.3389/fgene.2019.00793</a>.</li><li>[53] Q. Huang, J. Zhang, L. Wei, F. Guo, Q. Zou, 6mA-RicePred: a method for identifying DNA N6-methyladenine sites in the rice genome based on feature fusion, Front. Plant Sci. 11 (2020) 4, <a href=https://doi.org/10.3389/fpls.2020.00004>https://doi.org/10.3389/fpls.2020.00004</a>.</li><li>[54] Z. Liu, W. Dong, W. Jiang, Z. He, csDMA: an improved bioinformatics tool for identifying DNA 6 mA modifications via Chou’s 5-step rule, Sci. Rep. 9 (1) (2019) 1–9, <a href=https://doi.org/10.1038/s41598-019-49430-4>https://doi.org/10.1038/s41598-019-49430-4</a>.</li><li>[55] H. Xu, R. Hu, P. Jia, Z. Zhao, 6mA-Finder: a novel online tool for predicting DNA N6-methyladenine sites in genomes, Bioinformatics 36 (10) (2020) 3257–3259, <a href=https://doi.org/10.1093/bioinformatics/btaa113>https://doi.org/10.1093/bioinformatics/btaa113</a>.</li><li>[72] Q. Huang, W. Zhou, F. Guo, L. Xu, L. Zhang, 6mA-Pred: identifying DNA N6- methyladenine sites based on deep learning, PeerJ 9 (2021), e10813, <a href=https://doi.org/10.7717/peerj.10813>https://doi.org/10.7717/peerj.10813</a>.</li><li>[73] N.Q.K. Le, Q.T. Ho, Deep transformers and convolutional neural network in identifying DNA N6-methyladenine sites in cross-species genomes, Methods (2021), <a href=https://doi.org/10.1016/j.ymeth.2021.12.004>https://doi.org/10.1016/j.ymeth.2021.12.004</a>.</li><li>[74] M.U. Rehman, K.T. Chong, DNA6mA-MINT: DNA-6mA Modification Identification Neural Tool, Genes 11 (8) (2020) 898, <a href=https://doi.org/10.3390/genes11080898>https://doi.org/10.3390/genes11080898</a>.</li></ul></li></ul><h3 id=background>Background
<a class=anchor href=#background>#</a></h3><ul><li>What is methylation?
DNA methylation alters activities of DNA segments without changing the sequence, which thus yields a wide variety of roles in the cellular processes across organisms or tissues. In addition, the DNA methylation appears essential for normal development.</li><li>What is 6mA?
The 6mA refers to a biological process where the methyl group is attached to the 6-th nitrogen atom of adenine by the enzyme of DNA methyltransferase. The 6mA is a type of non-canonical DNA modification because it might occur in other nucleotide molecules including mRNA, tRNA, rRNA, small nuclear RNA (snRNA) as well as long non coding RNA</li><li>Why is it important?
A large volume of evidence suggested that the DNA 6mA would play vital roles in many key biological processes.
The DNA 6mA participated in regulation of gene expression both in prokaryotes and in some eukaryotes, and was responsible for DNA repair as well as DNA replication.
The 6mA distinguished invading foreign DNAs from host DNA in prokaryotes, was closely associated with many disease including tumor in human genome, and was involved in regulation of drug resistance in triple negative breast cancer</li></ul><h3 id=my-opinions>My Opinions
<a class=anchor href=#my-opinions>#</a></h3><p>For this paper, my perspective is this is a little bit trivial to solve the problem. For simplicity speaking, they just change the stacking model structure to a sequence structure. In addition, the result of this paper is exaggerating.</p><h3 id=comparison>Comparison
<a class=anchor href=#comparison>#</a></h3><ul><li><p>The result below is the experience on <code>6mA-rice-LV</code>(rice) dataset, and this paper method is <code>Deep6mAPred</code>. They used 5-fold cross validation on this data(<code>6mA-rice-LV</code>). In the original context, they said:</p><blockquote><p>The <code>Deep6mAPred</code> reached better Sn than three baseline methods (<code>Deep6mA</code> , <code>SNNRice6mA-large</code> and <code>Deep6mAPred</code>), and achieved competitive SP, ACC and MCC in contrast with the <code>Deep6mA</code>, which completely outperformed the <code>SNNRice6mA-large</code> and MM-6mAPred.</p></blockquote><p>However, the fun fact is the performance of $Sp, ACC, MCC, AUC$ is not good enough in this dataset.</p><p><img src=https://imgur.com/vccb3m0.png alt></p></li><li><p>The result below is for <code>6mA-rice-chen</code> dataset. Compared with <code>Deep6mA</code>, <code>Deep6mAPred</code> increased $Sn$ by 0.1572, ACC by 0.0750, MCC by 0.1436, and AUC of ROC curve by 0.0237, completely superior to the other two methods. The $Sp$ of <code>Deep6mAPred</code> is slightly lower than that of <code>Deep6mA</code>, but much higher than that of the other two methods.
<img src=https://imgur.com/ixsSXSK.png alt>
This result is quite distinguished that can show how special their model is under this another rice data.</p></li><li><p>This is ROC curves and PR curves result on <code>6mA-Fuse-R</code>(Rosa chinensis) and <code>6mA-Fuse-F</code>(Fragaria vesca, a kind of wild strawberry) respectively. In order to show how robust on their model, they try to test different species such as rose and wild strawberry without training, and the result is quite significant that almost similar to rice data.
<img src=https://imgur.com/eODMhQm.png alt></p></li><li><p>This is a self-created table that I wanna show the AUC of two curves with different species.
The original context said:</p><blockquote><p>As for the 6mA-Fuse-R, the <code>Deep6mAPred</code> outperformed three baseline methods in terms of the AUCs of ROC curves, while in terms of the AUCs of the PR curves it was equivalent to the <code>Deep6mA</code> but superior to the <code>SNNRice6mA-large</code> and <code>MM-6mAPred</code> a bit</p></blockquote><p>Follow the description above, we can know that the result of <code>6mA-Fuse-R</code> is better than three baseline methods but without any table or figure to prove that and this is not rigourous enough for this information.
<img src=https://imgur.com/UVaJ7sL.png alt></p></li><li><p>They also do some ablation experiment to prove that the attention mechanism they choose is quite valid and useful in this project.
We can see that in each experiment of different species, with attention mechanism is generally better than the experiment that without attention.
<img src=https://imgur.com/3Sj3Tfz.png alt></p></li></ul><h3 id=other-issue>Other Issue
<a class=anchor href=#other-issue>#</a></h3><ul><li>Why can wild rose and rice use the same architecture or we can ask how to process input data so that they can be applicable at the same model structure.</li><li>There is no extra explanation for the selected attention mechanism method.</li></ul><h3 id=conclusion>Conclusion
<a class=anchor href=#conclusion>#</a></h3><ul><li>The 6mA is a key mechanism of regulation in the cellular processes.</li><li>We presented a CNN and LSTM-based method (Deep6mAPred) with paralleling manner</li><li>They also used attention mechanism to improve their model and it&rsquo;s really helpful proved by ablation experiment.</li><li>In addition, they developed a user-friendly webserver to automatically detect 6mA sites</li><li>Deep6mAPred can also detect other plants</li></ul><h3 id=references>References
<a class=anchor href=#references>#</a></h3><h2 id=ensemble-learning-of-convolutional-neural-network-support-vector-machine-and-best-linear-unbiased-predictor-for-brain-age-prediction-aramis-contribution-to-the-predictive-analytics-competition-2019-challenge>Ensemble Learning of Convolutional Neural Network, Support Vector Machine, and Best Linear Unbiased Predictor for Brain Age Prediction: ARAMIS Contribution to the Predictive Analytics Competition 2019 Challenge
<a class=anchor href=#ensemble-learning-of-convolutional-neural-network-support-vector-machine-and-best-linear-unbiased-predictor-for-brain-age-prediction-aramis-contribution-to-the-predictive-analytics-competition-2019-challenge>#</a></h2><h3 id=introduction--motivation-1>Introduction & Motivation
<a class=anchor href=#introduction--motivation-1>#</a></h3><ul><li>Chronological age is an important risk factor for many conditions such as neurological disorders (e.g., Alzheimer’s and Parkinson’s), chronic (including cardiovascular) disorders, cancer, or stroke, to name a few.</li><li>However, it is an imperfect predictor of disease risk or of healthy individuals’ functional capability</li><li>Brain age (and PAD(predicted age difference)) trained on healthy participants may be applied to case-control samples where they have been shown to be non-specific predictors of disease status: Alzheimer’s disease and conversion[8-10], schizophrenia[11], alcohol dependence[12], cognitive impairment[13], or functional abilities[6, 14]</li><li>However, chronological age cannot explain everything, brain age correlates with disease, mortality, and function beyond what chronological age can explain</li></ul><h3 id=related-works-1>Related works
<a class=anchor href=#related-works-1>#</a></h3><p>[2]Cole J, Marioni RE, Harris SE, Deary IJ. Brain age and other bodily “ages”: implications for neuropsychiatry. Mol Psychiatr. (2019) 24:266–81. doi: 10.1038/s41380-018-0098-1
[3]Horvath S, Raj K. DNA methylation-based biomarkers and the epigenetic clock theory of ageing. Nat Rev Genet. (2018) 19:371–84. doi: 10.1038/s41576-018-0004-3
[4]Sajedi H, Pardakhti N. Age prediction based on brain MRI image: a survey. J Med Syst. (2019) 43:279. doi: 10.1007/s10916-019-1401-7
[5]Baker GT, Sprott RL. Biomarkers of aging. Exp Gerontol. (1988) 23:223–39. doi: 10.1016/0531-5565(88)90025-3
[6]Cole J, Ritchie SJ, Bastin ME, Valdés Hernández MC, Muñoz Maniega S, Royle N, et al. Brain age predicts mortality. Mol Psychiatr. (2018) 23:1385–92. doi: 10.1038/mp.2017.62
[8]Franke K, Gaser C. Longitudinal changes in individual brainAGE in healthy aging, mild cognitive impairment, alzheimer’s disease. GeroPsych. (2012) 25:235–45. doi: 10.1024/1662-9647/a000074
[9]Gaser C, Franke K, Klöppel S, Koutsouleris N, Sauer H. BrainAGE in mild cognitive impaired patients: predicting the conversion to alzheimer’s disease. PLoS ONE. (2013) 8:67346. doi: 10.1371/journal.pone.0067346
[10]Wang J, Knol MJ, Tiulpin A, Dubost F, Bruijne M, de, et al. Gray matter age prediction as a biomarker for risk of dementia. Proc Natl Acad Sci USA. (2019) 116:21213–8. doi: 10.1073/pnas.1902376116
[11]Koutsouleris N, Davatzikos C, Borgwardt S, Gaser C, Bottlender R, Frodl T, et al. Accelerated brain aging in schizophrenia and beyond: a neuroanatomical marker of psychiatric disorders. Schizophrenia Bull. (2014) 40:1140–53. doi: 10.1093/schbul/sbt142
[12]Guggenmos M, Schmack K, Sekutowicz M, Garbusow M, Sebold M, Sommer C, et al. Quantitative neurobiological evidence for accelerated brain aging in alcohol dependence. Transl Psychiatr. (2017) 7:1279. doi: 10.1038/s41398-017-0037-y
[13]Liem F, Varoquaux G, Kynast J, Beyer F, Kharabian Masouleh S, Huntenburg JM, et al. Predicting brain-age from multimodal imaging data captures cognitive impairment. NeuroImage. (2017) 148:179–88. doi: 10.1016/j.neuroimage.2016.11.005
[14]Beheshti I, Maikusa N, Matsuda H. The association between “Brain- Age Score” (BAS) and traditional neuropsychological screening tools in Alzheimer’s disease. Brain Behav. (2018) 8:e01020. doi: 10.1002/brb3.1020</p><h3 id=background-1>Background
<a class=anchor href=#background-1>#</a></h3><ul><li>Latest research focus on telomere length, methylation site, brain structure, and function[2-6]</li><li>In particular, brain age estimation from MRI images is a rapidly expanding field of research with several hundred publications to date</li></ul><h3 id=my-opinions-1>My Opinions
<a class=anchor href=#my-opinions-1>#</a></h3><p>The main opinion to this paper is that it&rsquo;s report of the competition they attended. And listed as clear as possible what problems they encountered, what techniques they used etc.</p><h3 id=comparison-1>Comparison
<a class=anchor href=#comparison-1>#</a></h3><ul><li><p>The * symbol represents a significant reduction in $MAE$ by Ensemble Learning compared to Inception alone ($p\ value &lt; 0.05$)</p><ul><li>For the objective of minimize MAE, the way of deep learning is better than <code>BLUP</code> and <code>SVM</code> ($pvalue\ of\ paired\ t-test&lt;3.1e-4$)</li><li>There was no significant difference in the performance of the deep learning algorithms ($p > 0.027$)</li><li>In contrast, Ensemble Learning&rsquo;s $MAE=3.46$, there is a significant difference (p=1.3e-4)</li><li>Taking challenge 2 as an example, the author uses median and mean absolute deviation per site to rescale the prediction. The results show that $MAE$ will increase by one year compared to the original one, but will reduce the bias. The same that ensemble learning has a significant improvement compared to Inception($p=0.010$).
<img src=https://imgur.com/5jR4hWo.png alt></li></ul></li><li><p>They also tried to evaluate whether their conclusions depend on the train/test split used in the previous section by performing a 5-fold cross-validation experiment.</p><ul><li>Within each fold, they found a nominally significant difference in MAE between <code>BLUP</code>/<code>SVM</code> and <code>ResNet</code> ($p &lt; 5.5E−3$)</li><li>In each fold, the composite age score using linear regression outperformed <code>Inception V1</code>&rsquo;s predictions ($p &lt; 0.0022$). For folds 2 and 3, ensemble learning via random trees significantly outperforms <code>Inception V1</code> alone ($p=4.0E−3 and 3.4E−4$)</li><li>Note that the $MAE$ obtained using Random Forest is very close to the $MAE$ obtained by taking the mean or median score for each person. We cannot conclude that there is a significant difference between <strong>linear model combinations</strong> and <strong>random forests</strong>.
<img src=https://imgur.com/zyIaGAd.png alt></li></ul></li><li><p>The low performance of <code>BLUP</code>/<code>SVM</code> shown above compared to deep learning algorithms motivated the authors to test whether it could be attributed to the input data or the algorithm itself. Therefore, the author retrains <code>BLUP</code> and <code>SVM</code> <font color=FF0000>(trained on gray matter maps)</font></p><ul><li>† Symbols represent: the algorithm trained with gray matter map is significantly <strong>better than</strong> the algorithm trained with surface-based vertices ($p &lt; 0.05/15$).</li><li>The * symbol indicates: the performance of the algorithm trained on the gray matter image is significantly <strong>lower than</strong> that of <code>Inception V1</code> ($p &lt; 0.05/15$)</li><li>Despite the reduction in MAE, <code>BLUP-mean</code> and <code>SVM</code> trained on gray matter still performed <font color=FF0000><strong>worse than</strong></font> <code>Inception V1</code> ($p &lt; 0.0033$), although the difference between <code>Inception V1</code> and <code>BLUP-quantile</code> became not significant.
<img src=https://imgur.com/gO86vVb.png alt></li></ul></li><li><p>The participant is older, the prediction error is larger. → Therefore, the predictor will tends to underestimate the age of older participants and overestimate the age of younger participants.
We did not observe significant associations of prediction errors with gender or location
<img src=https://imgur.com/a3ugXLy.png alt></p></li></ul><h3 id=other-issue-1>Other Issue
<a class=anchor href=#other-issue-1>#</a></h3><ul><li>They didn&rsquo;t explain why they used two <code>6-Layers CNN</code> to combine and the effect in detailed.</li><li>They also didn&rsquo;t explain the gray/white matter map difference and the properties of these maps in detailed.</li></ul><h3 id=conclusion-1>Conclusion
<a class=anchor href=#conclusion-1>#</a></h3><ul><li>Proposed an ensemble learning algorithm of 7 different age predictions from T1w MRI images</li><li>Ranked third in PAC2019</li><li>Ranking of prediction accuracy may be highly dependent on the metric chosen as well as on the test data</li><li>Evaluated the effect on performance of algorithm choice, ensemble learning methods, feature input/data processing, number and type of scores in ensemble learning, and covariates such as age, sex, and site</li></ul><h3 id=references-1>References
<a class=anchor href=#references-1>#</a></h3><h2 id=machine-learning-workflows-to-estimate-class-probabilities-for-precision-cancer-diagnostics-on-dna-methylation-microarray-data>Machine learning workflows to estimate class probabilities for precision cancer diagnostics on DNA methylation microarray data
<a class=anchor href=#machine-learning-workflows-to-estimate-class-probabilities-for-precision-cancer-diagnostics-on-dna-methylation-microarray-data>#</a></h2><h3 id=introduction--motivation-2>Introduction & Motivation
<a class=anchor href=#introduction--motivation-2>#</a></h3><ul><li>DNA methylation data-based precision cancer diagnostics</li><li>Application for class probability (CP): Stratified Medicine</li><li>Standards for choosing statistical methods with regard to well-calibrated probability estimates for these typically highly multiclass classification tasks are still lacking</li><li>We compared these workflows on a recently published brain tumor 450k DNA methylation cohort of 2,801 samples with 91 diagnostic categories using a 5 5-fold nested cross validation scheme and demonstrated their generalizability on external data from The Cancer Genome Atlas</li><li>Purpose of this study:<ul><li>to perform a benchmark analysis to support the choice for optimal DNA methylation microarray data analysis through extensive comparisons of well-established ML classifiers and their combination with post-processing algorithms</li><li>such as <code>Platt Scaling</code> and ridge-penalized <code>multinomial L</code>(MR)</li></ul></li><li><code>ELNET</code> was the top stand-alone classifier</li><li>The best overall two-stage workflow was MR-calibrated <code>SVM</code> with linear kernels closely followed by ridge-calibrated tuned RF</li><li>MR was the most effective regardless of the primary classifier</li><li>The number of features (p) vastly outnumbers the sample size (n)</li><li>Therefore, a more reasonable require predictions ment is that the estimated CP function provides well-calibrated(e.g. <code>Platt scaling</code>)</li><li>Multiclass classification problems → unbalanced classification problems</li></ul><h3 id=related-works-2>Related works
<a class=anchor href=#related-works-2>#</a></h3><ul><li>High multiclass & unbalanced classification problem
[7]Baek, S., Tsai, C.-A. & Chen, J. J. Development of biomarker classifiers from high-dimensional data. Brief.Bioinform. 10, 537–546 (2009).
[8]Dupuy, A. & Simon, R. M. Critical review of published microarray studies for cancer outcome andguidelines on statistical analysis and reporting. J. Natl Cancer Inst. 99,147–157 (2007).
[9]. Hastie, T., Tibshirani, R. & Friedman, J. The Elements of Statistical Learning: Data Mining, Inference andPrediction 2nd edn (Springer, New York, NY, 2009).
[10]. Lee, J. W., Lee, J. B., Park, M. & Song, S. H. An extensive comparison of recent classification tools applied tomicroarray data. Comput. Stat. Data Anal. 48, 869–885 (2005).
[11]. Simon, R. Roadmap for developing and validating therapeutically relevant genomic classifiers. J. Clin. Oncol.23, 7332–7341 (2005).</li><li>DNA methylation application
[1]. Capper, D. et al. DNA methylation-based classification of central nervous system tumours. Nature 555,469–474 (2018).
[4]. Rodríguez-Paredes, M. & Esteller, M. Cancer epigenetics reaches mainstream oncology. Nat. Med. 17,330–339 (2011).
[13]. Fernandez, A. F. et al. A DNA methylation fingerprint of 1628 human samples. Genome Res. 22, 407–419 (2012).
[14]. Wiestler, B. et al. Assessing CpG island methylator phenotype, 1p/19q codeletion, and MGMT promotermethylation from epigenome-wide data in the biomarker cohort of the NOA-04 trial. Neuro Oncol. 16,1630–1638 (2014).</li><li>The number of features (p) vastly outnumbers the sample size (n)
[9]. Hastie, T., Tibshirani, R. & Friedman, J. The Elements of Statistical Learning: Data Mining, Inference and Prediction 2nd edn (Springer, New York, NY, 2009).
[20]. Simon, R. Class probability estimation for medical studies. Biom. J. 56, 597–600 (2014).</li></ul><h3 id=background-2>Background
<a class=anchor href=#background-2>#</a></h3><ul><li>The problem of stratified medicine<ol><li>Unbiased classification problem (there are fewer patients in some categories, of course there will be inequality problems)</li><li>Usually medical diagnosis is to classify patients in several categories of interest, but real medical behavior should be able to achieve multi-category diagnosis (50 or more than 100 categories)</li></ol></li></ul><h3 id=my-opinions-2>My Opinions
<a class=anchor href=#my-opinions-2>#</a></h3><p>This paper is just like a <code>Readme</code> file that wanna teach someone how to use their tool, each technique they used, each problem they encountered, and also which programming package they used etc. as clear as possible. Although the paper should be as clear as possible, but too much unnecessary information is really a waste of time and annoying.</p><h3 id=comparison-2>Comparison
<a class=anchor href=#comparison-2>#</a></h3><ul><li><p><strong>Random Forest(RFs)</strong></p><ul><li>Vanilla RF(vRF)<ul><li>The ME of vRF was 4.8%, the AUC was 99.9%, and the corresponding BS and LL were 0.32 and 0.78, respectively</li><li>Platt scaling with LR and <code>FLR</code> improves BS and LL by a factor of 2-4, furthermore, <code>FLR</code> is better than <code>LR</code></li><li>MR slightly outperformed Platt&rsquo;s two variants and achieved very low 10th and 9th overall BS (0.073) and LL (0.155) metrics respectively</li></ul></li><li>tuned RF(tRF)<ul><li>RF tuned for ME (<code>tRFME</code>) showed 10th overall error rate (3.5%) and 4th AUC (99.9%), while it had relatively high BS (0.35) and LL (0.86) similar to <code>vRF</code></li><li>Both <code>tRFBS</code> and <code>tRFLL</code> have higher error rates, about 5.5%</li><li>After calibration with <font color=FF0000><strong>MR</strong></font>, almost all versions of <code>tRF</code> get the biggest performance improvement</li></ul></li></ul><p><img src=https://imgur.com/MgjyT0T.png alt></p></li><li><p><strong>ELNET</strong></p><ul><li>It used 1,000 most variable CpG probes</li><li>ME ranked 8th, AUC ranked 5th</li><li>ME (2.7%), BS (0.048) and LL (0.109) and negligibly low AUC (99.9 %)</li></ul><p><img src=https://imgur.com/A2RRIAp.png alt></p></li><li><p><strong>SVM</strong></p><ul><li>More effective ME = 2.1% (lowest overall) with Platt scaling with Firth regression</li><li>While simple LR can be more effective to improve BS (second) and LL (fourth) by 8-9 times respectively</li><li>MR (<font color=FF0000><strong><code>SVM-LK+MR</code></strong></font>) achieves the most comprehensive improvement across all metrics. It reduced BS by a factor of 9.5 and LL by a factor of 11.5, resulting in the second lowest ME (2.1%) and AUC (99.9%), lowest BS (0.039) and lowest LL (0.085)</li></ul><p><img src=https://imgur.com/CaeII1m.png alt></p></li><li><p><strong>Boost Tree</strong></p><ul><li>Boosted model using ME as evaluation metric outperforms model using LL</li><li>Overall ME of 5.1% and AUC of 99.9%, with the second lowest BS (0.15) and LL (0.43) among the base ML classifiers studied</li></ul><p><img src=https://imgur.com/XXdGN2p.png alt></p></li></ul><h3 id=other-issue-2>Other Issue
<a class=anchor href=#other-issue-2>#</a></h3><h3 id=conclusion-2>Conclusion
<a class=anchor href=#conclusion-2>#</a></h3><ul><li>We performed extensive comparative analyses of four well-established classifier algorithms including RF, <code>ELNET</code>, <code>SVM</code> and boosted ensemble trees in combination with <code>Platt scaling</code> and <code>multinomial ridge regression</code></li><li>The best overall two-stage workflow was MR-calibrated <code>SVM-LK</code>, and it generated the best overall BS, LL and AUC metrics.</li><li>For calibration, <code>multinomial ridge-penalized regression</code> was the most effective regardless of the primary classifier</li></ul><h3 id=references-2>References
<a class=anchor href=#references-2>#</a></h3></article><footer class=book-footer><div class="flex flex-wrap justify-between"></div><script>(function(){function e(e){const t=window.getSelection(),n=document.createRange();n.selectNodeContents(e),t.removeAllRanges(),t.addRange(n)}document.querySelectorAll("pre code").forEach(t=>{t.addEventListener("click",function(){if(window.getSelection().toString())return;e(t.parentElement),navigator.clipboard&&navigator.clipboard.writeText(t.parentElement.textContent)})})})()</script></footer><div class=book-comments></div><label for=menu-control class="hidden book-menu-overlay"></label></div><aside class=book-toc><div class=book-toc-content><nav id=TableOfContents><ul><li><ul><li></li></ul></li><li><a href=#deep6mapred-a-cnn-and-bi-lstm-based-deep-learning-method-for-predicting-dna-n6-methyladenosine-sites-across-plant-species>Deep6mAPred: A CNN and Bi-LSTM-based deep learning method for predicting DNA N6-methyladenosine sites across plant species</a><ul><li><a href=#introduction--motivation>Introduction & Motivation</a></li><li><a href=#related-works>Related works</a></li><li><a href=#background>Background</a></li><li><a href=#my-opinions>My Opinions</a></li><li><a href=#comparison>Comparison</a></li><li><a href=#other-issue>Other Issue</a></li><li><a href=#conclusion>Conclusion</a></li><li><a href=#references>References</a></li></ul></li><li><a href=#ensemble-learning-of-convolutional-neural-network-support-vector-machine-and-best-linear-unbiased-predictor-for-brain-age-prediction-aramis-contribution-to-the-predictive-analytics-competition-2019-challenge>Ensemble Learning of Convolutional Neural Network, Support Vector Machine, and Best Linear Unbiased Predictor for Brain Age Prediction: ARAMIS Contribution to the Predictive Analytics Competition 2019 Challenge</a><ul><li><a href=#introduction--motivation-1>Introduction & Motivation</a></li><li><a href=#related-works-1>Related works</a></li><li><a href=#background-1>Background</a></li><li><a href=#my-opinions-1>My Opinions</a></li><li><a href=#comparison-1>Comparison</a></li><li><a href=#other-issue-1>Other Issue</a></li><li><a href=#conclusion-1>Conclusion</a></li><li><a href=#references-1>References</a></li></ul></li><li><a href=#machine-learning-workflows-to-estimate-class-probabilities-for-precision-cancer-diagnostics-on-dna-methylation-microarray-data>Machine learning workflows to estimate class probabilities for precision cancer diagnostics on DNA methylation microarray data</a><ul><li><a href=#introduction--motivation-2>Introduction & Motivation</a></li><li><a href=#related-works-2>Related works</a></li><li><a href=#background-2>Background</a></li><li><a href=#my-opinions-2>My Opinions</a></li><li><a href=#comparison-2>Comparison</a></li><li><a href=#other-issue-2>Other Issue</a></li><li><a href=#conclusion-2>Conclusion</a></li><li><a href=#references-2>References</a></li></ul></li></ul></nav></div></aside></main></body></html>